mne/export/_edf.py

```diff
@@ -204,7 +205,9 @@ def _export_raw(fname, raw, physical_range, add_ch_type):
 
     for desc, onset, duration, ch_names in zip(
         raw.annotations.description,
-        raw.annotations.onset,
+        # subtract raw.first_time because EDF marks events starting from the first
+        # available data point and ignores raw.first_time
+        _sync_onset(raw, raw.annotations.onset, inverse=False),
```

Comment:
Also re-wrote the tests and they are passed without an issue now :D From my side it's in principle good to go.

One final note though is I'm using `_sync_onset`, which in addition to performing `annot_start = onset - raw._first_time` also `assert raw.info["meas_date"] == raw.annotations.orig_time`. Therefore, this would enforce users to export only `Raw` that has identical `meas_date` and `orig_time`.